### PR TITLE
Remove legacy logic for adding empty eth row

### PR DESCRIPTION
**Pull request by osdnk**
Possibly closes https://linear.app/rainbow/issue/RNBW-1589/when-switching-wallets-often-see-0-eth-coin-row-instead-of-loading

I noticed that we have logic for adding empty eth row in two places - in `buildWalletSections` and `helpers/assets`. The first one is performing some weird array modification that can be responsible for flickering. It looks like (See wallet `0x7B4C0C1b3a8a88a1fE00ABD7E96ADdcB4f330332`) this works correctly - I tested this.

I was not able to notice the bug.

### CODE REVIEW

File: src/helpers/buildWalletSections.js [hunk 1]
```diff
@@ -79,60 +78,14 @@ const filterWalletSections = sections =>
...
-        },
-      },
-      price: assets.eth.price,
-      symbol: 'ETH',
-      type: 'token',
-      uniqueId: 'eth',
-    };
-
-    if (section.data.length === 1) {
-      section.data.unshift(zeroEthRow);
```
Comment (osdnk) [inline src/helpers/buildWalletSections.js:118]: I think this is rly bad

**Comment by skylarbarrera** [comment]
looks like the logic in the buildWalletSection was to handle the scenario where the user has an NFT but no eth/erc-20's

https://cloud.skylarbarrera.com/Screen-Shot-2021-09-14-09-02-19.26.png

**Comment by mikedemarais** [comment]
> looks like the logic in the buildWalletSection was to handle the scenario where the user has an NFT but no eth/erc-20's
> 
> https://cloud.skylarbarrera.com/Screen-Shot-2021-09-14-09-02-19.26.png

this indeed why was why we added this `empty eth row`,,, 
so like we should definitely add a comment here explaining we wanna show it in `scenario where the user has an NFT but no eth/erc-20's` 

but as far as the grossness of the array manipulation,, @osdnk do we wanna quickly improve it or does this PR like get closed after we add a comment to the codeblock?

**Review by mrbaguvix — COMMENTED** [review]
IMO, Not sure we need to remove codeblocks

**Comment by osdnk** [comment]
todo: check with Skylar's dad account

**Review by skylarbarrera — APPROVED** [review]
lgtm

**Pull request merged**